### ISSUE TITLE
Public initializers for nested fragments

### ISSUE THREAD

**Issue by ni4ka** [apollographql/apollo-ios#611]
After update to Apollo 0.10.0 / 0.10.1 public initializers for nested fragments had changed.
Example of fragments below 

```
fragment MiniProductFrag on Product {
  id
  name
  ...
}

fragment ProductFrag on Product {
  ...MiniProductFrag
  bundle {
    products {
      ...MiniProductFrag
    }
  }
}
```

And the query 

```
query Products($productId: [ID]!) {
  products(ids: $productId) {
    ...ProductFrag
  }
}
```

In my ApolloGraphAPI.swift file previously `ProductFrag` had `public init(id: GraphQLID, name: String, ...)` but this initializer was removed and only `public init(unsafeResultMap: ResultMap)` left.
I used `ProductFrag(id:name:)` before but now I see an error `Extra argument 'id' in call`. What should I do?

**Comment by designatednerd** [comment]
It looks like the `id` might be something internal to what we're doing - where were you getting that `id` previously?

**Comment by ni4ka** [comment]
Mostly, we are using those initializers for tests, but also we can get `id` from the other request and use the `ProductFrag` with just a few fields for the query to get the full product info

**Comment by amol-c** [comment]
@designatednerd We require to create mock objects for unit testing. There is a related issue in the previous version of Apollo with the same issue that was resolved. Please find it below.

https://github.com/apollographql/apollo-ios/issues/52

**Comment by designatednerd** [comment]
I'll need to dive into some of the codegen on this one - that seems to be where the change was made.

**Comment by ni4ka** [comment]
Thank you! Can you please specify when approximately you can get back to us regarding this?

**Comment by designatednerd** [comment]
I don't have an exact timeline (partly because I am The Suck at Typescript, where everything is presently generated) but I'll see what I can figure out this week.

**Comment by designatednerd** [comment]
Heads up: I was not able to get a good line on making this work with JS, but it's definitely high priority for the Swift Codegen stuff I'm gonna be working on. 

I'd be happy to review a PR on [apollo-tooling](https://github.com/apollographql/apollo-tooling) for it though!

**Comment by nsmag** [comment]
We're facing this problem too. Some of nested fragments don't have those public initializers.

Is there any workaround for this?

**Comment by designatednerd** [comment]
Not at the moment but if you take a look at #939 you'll see I'm planning to add these as part of the swift codgen rewrite

**Comment by AnthonyMDev** [comment]
This issue is from a long time ago, but I wanted to give an update before closing it.

We've just released Apollo iOS 1.0 Alpha 5, which provides support for generated typed mocks for unit testing! Though we aren't providing initializers directly on your `SelectionSet` or `Fragment` objects, these test mocks can be used to support this case for unit testing.

Since the new 1.0 version will provide this support, I'm going to close this issue now. We would love to get your feedback on the 1.0 Alpha!
